src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -98,12 +98,6 @@ public DefaultTokenCacheStore(Context context) {
         if (mPrefs == null) {
             throw new IllegalStateException(ADALError.DEVICE_SHARED_PREF_IS_NOT_AVAILABLE.getDescription());
         }
-
-        try {
-            getStorageHelper().loadSecretKeyForAPI();
```

Comment:
I would leave loadSecretKeyForEncryption() here, because if secret key wasn't set dev should know it before sign in process. And this load can be removed from encryption